### PR TITLE
f32: document Sqrt as IEEE-754 correctly-rounded and add fused AbsPow34 (#183)

### LINKED ISSUE #183: f32: add a fused, correctly-rounded three-quarter power |x|^(3/4) (and document Sqrt's rounding)
### What

A fused elementwise primitive for the three-quarter power law:

```go
func f32.AbsPow34(dst, src []float32) // dst[i] = |src[i]|^(3/4), via sqrt(|x| * sqrt(|x|))
```

The `sqrt(x * sqrt(x))` identity yields `x^(3/4)` with two square roots and one multiply, no `pow` / `exp` / `log`, so it can be bit-exact rather than approximate.

### Why not just `f32.Pow(dst, src, 0.75)`

`Pow` is documented as the fused `exp(exp*ln(x))` kernel with relative error ~1.4e-5 on the SIMD path. That is fine for perceptual work but not for a bit-exact port that must match a C reference sample-for-sample. The C this mirrors literally computes `sqrtf(a * sqrtf(a))`, so a fused primitive on hardware sqrt would match it exactly.

### The real dependency: a correctly-rounded `Sqrt` contract

For either a fused kernel or a hand-composed `Abs -> Sqrt -> Mul -> Sqrt` to be provably bit-exact, `f32.Sqrt` needs to be the IEEE-754 correctly-rounded single-precision square root (which `VSQRTPS` / `FSQRT` already are). The current `f32.Sqrt` doc does not state its rounding. If you would rather not add `AbsPow34` at all, then just documenting `f32.Sqrt` as correctly-rounded is enough: callers can compose the four ops and rely on bit-exactness, paying only the extra memory passes a fused kernel would save.

### Why it is generic

The 3/4 power law is the loudness / companding curve in AAC and MP3 quantization and appears in psychoacoustic loudness models generally. It sits alongside the existing `Sqrt`, `Reciprocal`, `Pow` elementwise family, and bit-exactness against the pure-Go reference is already part of this library's promise (per the i16/i32 docs), which this primitive can honor where `Pow` cannot.

### Priority

Low. In my AAC encoder this leaf (`abs_pow34`) is ~2% of encode and is composable from existing ops today if `Sqrt` is guaranteed correctly-rounded. The core ask is really the `Sqrt` rounding guarantee; the fused `AbsPow34` is a convenience on top of it.

### PR DESCRIPTION
**Pull request by tphakala**
Closes #183.

Two deliverables.

**1. Documents `f32.Sqrt` as IEEE-754 correctly-rounded** (round-to-nearest-even) on every backend. `VSQRTPS`/`SQRTPS` (amd64) and `FSQRT` (arm64) are correctly-rounded by the ISA. The Go fallback computes `float32(math.Sqrt(float64(x)))`; that double rounding is provably innocuous for square root because binary64's 53-bit significand exceeds the `2*24+2 = 50` bits the classical result requires for `+ - * / sqrt`, so it returns the correctly-rounded binary32 result for every input. This lets callers compose bit-exact higher powers.

**2. Adds `f32.AbsPow34(dst, src []float32)`:** `dst[i] = |src[i]|^(3/4)` via `sqrt(|x| * sqrt(|x|))`, mirroring the C `sqrtf(a * sqrtf(a))` bit-for-bit. Backends: amd64 AVX (`VANDPS`/`VSQRTPS`/`VMULPS`/`VSQRTPS`) and SSE, arm64 NEON (hand-encoded `FABS`/`FSQRT`/`FMUL`/`FSQRT`), and a Go fallback. Each element is three single correctly-rounded float32 ops (no add, so the no-FMA contract is not in play), so the result is bit-identical across all backends with no relaxed tier, and `dst` may alias `src`.

**Contract note:** the intermediate `|x| * sqrt(|x|)` is a float32 multiply, so like the C form it saturates to `+Inf` for `|x|` above roughly `2^85.33` and underflows to `0` for very small `|x|`, even though the true `|x|^(3/4)` would be representable. This is intentional and matches the reference sample-for-sample; callers needing the full-range 3/4 power should use the approximate `Pow`. The Go reference keeps the middle multiply in float32 so it reproduces this exactly.

**Tests:** `TestSqrt_CorrectlyRounded` and `TestAbsPow34` validate bit-for-bit against an independent `math/big.Float` oracle over dense sweeps (bands around 1.0, powers of two and four, subnormals, near-midpoints, the overflow/underflow thresholds, `MaxFloat32`, `SmallestNonzeroFloat32`, `0`/`+Inf`/`NaN`/`-0.0`), plus alloc-free, tail-untouched, clamp, aliasing, unaligned-operand, dispatch-pin, and differential-fuzz coverage, and a `BenchmarkAbsPow34`.

**Hardware verification:** verified bit-exact on real hardware. AVX2 (i7-1260P) ran the AVX and SSE kernels; NEON (Raspberry Pi 5) ran the `FSQRT`/NEON kernels; both matched the `big.Float` oracle across the full sweep, allocs = 0.

Note for reviewers: the amd64 dispatch for AbsPow34 uses a direct `cpu.X86.AVX`/`SSE2` switch rather than the function-pointer table, matching the exp/relu/pow family, because an indirect call through a func value defeats `//go:noescape` and forced 2 allocs/run, breaking the zero-alloc contract.

## Summary by CodeRabbit

* **New Features**
  * Added `AbsPow34` for element-wise 3/4-power magnitude calculations on float32 data.
  * Added optimized support across supported CPU architectures with automatic fallback behavior.
  * Expanded `Sqrt` documentation to describe its correctly rounded IEEE-754 results.

* **Tests**
  * Added comprehensive correctness, edge-case, aliasing, allocation, dispatch, and fuzz coverage.
  * Added validation for correctly rounded square-root results.

* **Benchmarks**
  * Added throughput benchmarks comparing optimized and Go implementations.

### CODE REVIEW

File: f32/sqrt_rounding_test.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	default:
+		return bigSqrtToF32(x)
+	}
+}
+
+// sqrtSweepInputs builds the representative input sweep from the design: specials
+// (0, -0, +Inf, NaN), perfect squares, all powers of two and four, a dense
+// contiguous band of float32 bit patterns around 1.0, a full-range exponent
+// sweep, subnormals, constructed near-midpoint targets, and random draws across
+// the whole positive range. Exhaustive portions shrink under -short.
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/sqrt_rounding_test.go]: Fixed in aa84e3c. Corrected the doc and block comments: the loop appends every small integer and half-integer up to 2048, of which the perfect squares have exact roots and the rest have irrational roots that exercise the rounding decision.

File: f32/sqrt_rounding_test.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+func sqrtSweepInputs(short bool) []float32 {
+	in := make([]float32, 0, 1<<21)
+
+	// Specials and small perfect squares (exact roots).
+	in = append(in, 0, float32(math.Copysign(0, -1)), float32(math.Inf(1)), float32(math.NaN()))
+	for k := 0; k <= 2048; k++ {
+		in = append(in, float32(k), float32(k)*0.5) // perfect squares and half-integers
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply f32/sqrt_rounding_test.go]: Fixed in aa84e3c. Reworded the inline comment to 'all integers and half-integers 0..2048'; only the perfect squares among them have exact roots.
